Makefile

```diff
@@ -118,6 +118,18 @@ shellcheck/install: $(BIN_DIR) guard/program/xz
 	rm -rf $(@D)-*
 	$(@D) --version
 
+# For editorconfig-checker, the 'curl' command needs the options to follow
+# redirects (e.g., "-L -C - ").  The tar file consists of a single file,
+# ./bin/ec-linux-amd64.
+ec/install: EC_BASE_NAME := ec-linux-$(ARCH)
+ec/install:
+	@ echo "[$@]: Installing $(@D)..."
+	$(CURL) -s -O -L -C - https://github.com/editorconfig-checker/editorconfig-checker/releases/latest/download/$(EC_BASE_NAME).tar.gz
```

Comment:
I'd suggest using the target `stream/gh-release`, to support getting a specific version and ensure we do not hit rate limits by making unauthenticated calls to github endpoints...

```
ec/install: EC_VERSION ?= latest
ec/install: EC_BASE_NAME ?= ec-$(OS)-$(ARCH)
ec/install:
        @ echo "[$@]: Installing $(@D)..."
        $(MAKE) -s stream/gh-release/$(@D) OWNER=editorconfig-checker REPO=editorconfig-checker VERSION=$(EC_VERSION) QUERY='.name | endswith("$(EC_BASE_NAME).tar.gz")' | tar -C "$(BIN_DIR)" --strip-components=1 -xzvf -
        ln -sf "$(BIN_DIR)/$(EC_BASE_NAME)" "$(BIN_DIR)/$(@D)"
        $(@D) --version
        @ echo "[$@]: Completed successfully!"
```

Reply:
I got the only line that required intelligence wrong; the remaining changes are just busywork.  Loren, I would have preferred it if you referred me to stream/gh-release and let me figure out the actual line.  I did not know what the stream/gh-release did and I should have asked somebody.

Reply:
Noted, my apologies! :)

Reply:
Ok. Then, how about switching stream/gh-release from a make target to a macro, `stream_github_release`, a la `download_github_release`? Looking at it now, I don't know why I didn't do that originally... ;)

Reply:
I didn't know about the issue of rate limits for unauthenticated calls to Git endpoints.  I had learned about GitHub Releases and the Git API, and I had tried some of the commands from the Makefile on the command line but could not get them to work.  I think one of the issues was that I used "mstruebing" as the repo owner instead of "editorconfig-checker".  

When I finally got a command to work I didn't see why I should use any of the other commands that I couldn't get to work.  Also, there are a variety of curl commands in the Makefile and not a lot of consistency.

Perhaps someone else could take a look at the variety of curl commands in the Makefile and at using a macro for stream_github_release and document why this macro is beneficial.  Also, the CURL definition has an argument of "-sSL" ... I think they meant either "-s" or "--ssl" or maybe both?

Reply:
Hmm, I only see one definition of `$(CURL)` and every target is using it, no random `curl` usage?

I mentioned the macro `stream_github_release` only as it would replace the stream/gh-release target, and you seemed disappointed I provided the whole command instead of letting you figure it out. 😁 It's not critical, things work as is, but the macro I think would be a little cleaner as it wouldn't require re-invoking `$(MAKE)`. Basically, it's a way to improve what I gave you...

Reply:
There is only one definition of CURL and a variety of ways it's used in the Makefile.